### DISCUSSION TITLE
add zpush to iredmail

### DISCUSSION THREAD

**Discussion by orlovnv** [Z-Hub/Z-Push#122] [Q&A] [answered]
hello, I try to install zpush to iredmail latest.
all good besides one error 
 [error] 2796#2796: *551 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error: Call to undefined function parse_ini_file() in /usr/local/lib/z-push/lib/core/zpush.php:283
Stack trace:
#0 /usr/local/lib/z-push/index.php(45): ZPush::CheckConfig()
#1 {main}
  thrown in /usr/local/lib/z-push/lib/core/zpush.php on line 283" while reading response header from upstream, client: 192.168.2.37, server: _, request: "OPTIONS /Microsoft-Server-ActiveSync HTTP/1.1", upstream: "fastcgi://127.0.0.1:9999", host:
my nginx tmpl:

       location /Microsoft-Server-ActiveSync {
                include /etc/nginx/fastcgi_params;
                fastcgi_param SCRIPT_FILENAME /usr/local/lib/z-push/index.php;
                fastcgi_read_timeout 630;
                include /etc/nginx/templates/fastcgi_php.tmpl;
                client_max_body_size 128M;
        }
        location ~* ^/autodiscover/autodiscover.xml$ {
                include /etc/nginx/fastcgi_params;
                fastcgi_param SCRIPT_FILENAME /usr/local/lib/z-push/autodiscover/autodiscover.php;
                include /etc/nginx/templates/fastcgi_php.tmpl;        
} 

autodiscover/autodiscover.xml popup login window , and /Microsoft-Server-ActiveSync doesn't I see clear window

**Comment by matidau** [comment]
You might want to check your php installation, parse_ini_file is a php function

https://www.php.net/manual/en/function.parse-ini-file.php

**Reply by orlovnv** [reply]
ok now new error 
![Снимок экрана 2024-11-23 в 10 45 25](https://github.com/user-attachments/assets/ed5119ff-fef8-49a2-aa0d-80d09bab6071)

**Reply by orlovnv** [reply]
I even try chmod 777 to that folders

**Reply by orlovnv** [reply]
im sorry now it all is good, but I have to chmod 777 /var/lib/z-push. may be owner of that folders should be not root?

**Reply by matidau** [reply]
Depending on your web server

```
chown www-data:www-data /var/lib/z-push
```

**Comment by orlovnv** [comment]
one more question why imap mail.app stop working I mean native mail app on makbook

**Reply by matidau** [reply, marked as answer]
I'm not sure, I don't use macOS. Maybe someone else can help.

**Comment by orlovnv** [comment]
it was my fault fial2ban block ip
